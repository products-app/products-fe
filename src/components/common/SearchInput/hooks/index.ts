import { useStore } from '@/stores'

const useSearch = () => {
  const { searchTerm, setSearch } = useStore()

  const handleChange = (value: string) => {
    setSearch(value)
  }

  return {
    searchTerm,
    handleChange,
  }
}

export { useSearch }
