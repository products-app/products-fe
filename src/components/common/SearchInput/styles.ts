import { styled } from '@lebernardo/react'

export const InputContainer = styled('form', {
  flex: '1',
  maxWidth: '20rem',
})
