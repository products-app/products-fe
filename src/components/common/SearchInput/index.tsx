import { TextInput } from '@lebernardo/react'
import { useSearch } from './hooks'
import { InputContainer } from './styles'

const SearchInput = () => {
  const { searchTerm, handleChange } = useSearch()

  return (
    <InputContainer>
      <TextInput
        type="text"
        id="search"
        placeholder="Pesquise por produtos"
        value={searchTerm}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          handleChange(e.target.value)
        }
      />
    </InputContainer>
  )
}

export default SearchInput
