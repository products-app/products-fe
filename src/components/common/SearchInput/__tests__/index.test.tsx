import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import SearchInput from '../index'

const props = {
  searchTerm: '',
  handleChange: jest.fn(),
  placeholder: 'test placeholder',
}

const mockSearchValue = 'Product 1'

describe('Search component', () => {
  afterEach(() => {
    jest.restoreAllMocks()
  })

  test('should render correctly', () => {
    const { container } = render(<SearchInput {...props} />)

    expect(screen.getByRole('textbox')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('should change search input value and press enter', async () => {
    render(<SearchInput {...props} />)

    const input = screen.getByRole('textbox') as HTMLInputElement
    await userEvent.type(input, mockSearchValue)

    await userEvent.keyboard('{Enter}')

    expect(input).toHaveValue(mockSearchValue)
  })

  test('should change search input value and press backspace', async () => {
    render(<SearchInput {...props} />)

    const input = screen.getByRole('textbox') as HTMLInputElement
    await userEvent.type(input, 'a')

    await userEvent.keyboard('{Backspace}')

    expect(input).toHaveValue('')
  })
})
