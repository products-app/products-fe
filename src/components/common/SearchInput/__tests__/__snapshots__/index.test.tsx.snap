// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render correctly 1`] = `
<div>
  <div
    class="c-eGMMdN"
  >
    <div
      class="c-bKZIAC"
    >
      <input
        class="c-bgjSgL"
        id="search"
        placeholder="test placeholder"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
